src/geosearch.ts

```diff
@@ -59,97 +69,42 @@ export class GeoSearcher {
-            });
+        let params: searchProviderParms = {
+            query: query,
+        };
+
+        if (this.usingGooglePlacesSearch() && searchArea) {
+            const centerOfSearch = searchArea?.getCenter();
+            params.location = `${centerOfSearch.lat},${centerOfSearch.lng}`;
+        }
+        let searchResults = await this.searchProvider.search(params);
+
+        if (!this.usingGooglePlacesSearch()) {
             searchResults = searchResults.slice(
                 0,
                 consts.MAX_EXTERNAL_SEARCH_SUGGESTIONS
             );
-            results = results.concat(
-                searchResults.map(
-                    (result) =>
-                        ({
-                            name: result.label,
-                            location: new leaflet.LatLng(result.y, result.x),
-                            resultType: 'searchResult',
-                        } as GeoSearchResult)
-                )
-            );
         }
 
+        results = results.concat(
+            searchResults.map(
+                (result) =>
+                    ({
+                        name: result.label,
+                        location: new leaflet.LatLng(result.y, result.x),
+                        resultType: 'searchResult',
+                    } as GeoSearchResult)
+            )
+        );
+
         return results;
     }
-}
 
-export async function googlePlacesSearch(
-    query: string,
-    settings: PluginSettings,
-    centerOfSearch: leaflet.LatLng | null
-): Promise<GeoSearchResult[]> {
-    if (settings.searchProvider != 'google' || !settings.useGooglePlaces)
-        return [];
-    const googleApiKey = settings.geocodingApiKey;
-    const params = {
-        query: query,
-        key: googleApiKey,
-    };
-    if (centerOfSearch)
-        (params as any)[
-            'location'
-        ] = `${centerOfSearch.lat},${centerOfSearch.lng}`;
-    const googleUrl =
-        'https://maps.googleapis.com/maps/api/place/textsearch/json?' +
-        querystring.stringify(params);
-    const googleContent = await request({ url: googleUrl });
-    const jsonContent = JSON.parse(googleContent) as any;
-    let results: GeoSearchResult[] = [];
-    if (
-        jsonContent &&
-        'results' in jsonContent &&
-        jsonContent?.results.length > 0
-    ) {
-        for (const result of jsonContent.results) {
-            const location = result.geometry?.location;
-            if (location && location.lat && location.lng) {
-                const geolocation = new leaflet.LatLng(
-                    location.lat,
-                    location.lng
-                );
-                results.push({
-                    name: `${result?.name} (${result?.formatted_address})`,
-                    location: geolocation,
-                    resultType: 'searchResult',
-                } as GeoSearchResult);
-            }
-        }
+    usingGooglePlacesSearch(): Boolean {
```

Comment:
This should be the Typescript type `boolean` and not the ES5 interface `Boolean`.